src/pulse3D/metrics.py

```diff
@@ -317,12 +319,13 @@ def calculate_twitch_widths(
             width_df: DataFrame, where each index is an integer representing the time points, and each column is a      percent-twitch width of all the peaks of interest
             coordinate_df: MultiIndex DataFrame, where each index is an integer representing the time points, and each  column level corresponds to the time (X) / force(Y), contration (rising) / relaxation (falling), and percent-twitch width coordinates
         """
-        width_df = pd.DataFrame(index=list(twitch_indices.index), columns=list(np.arange(10, 95, 5)))
-
-        columns = pd.MultiIndex.from_product(
-            [["force", "time"], ["contraction", "relaxation"], list(np.arange(10, 95, 5))]
-        )
-        coordinate_df = pd.DataFrame(index=list(twitch_indices.index), columns=columns)
+        
+        coordinate_dict = {h: {k: {i: {j: None 
```

Comment:
Should use descriptive variable names instead of h,i,j,k

Reply:
updated key names